src/components/VmDetail/index.js

```diff
@@ -70,6 +76,9 @@ class VmDetail extends Component {
     const icon = icons.get(iconId)
     const disks = vm.get('disks')
 
+    const onToggleRenderDisks = () => { this.setState({ renderDisks: !this.state.renderDisks }) }
```

Comment:
That's a good question, I was considering as well.

Of course, this redux principle works and has its benefits.

On the other hand I don't think we shall follow it blindly since for small and purely local changes this approach seems to be overkilling, causing a lot of computation and boilerplate code.
Moreover, the Redux tree becomes flooded by a lot of boolean-like mess leading to inefficiency in code readability same as the data manipulation.

A good example for this kind of "exception" is the "Button" component in VmActions - it handles part of its state locally to render the user confirmation.
Please consider the boiler plate code needed to let this happen via Redux. It would slowly but clearly increase the complexity of the application without benefits.

Maybe the VmDetail is not a good example of that - I can imagine (but not used recently) the "hide/show" to be managed externally to the component.
So yes, from long-term perspective, Redux here might make sense (anyway, refactoring is still posiible).

Let's wrap the VmDisks by ExpandableVmDisks handle this state. It would be a hook for either refresh or moving the state to Redux